GME/Gme/GMEApp.cpp

```diff
@@ -2972,7 +3170,73 @@ void CGMEApp::OnRunInterpreter(UINT nID) {
 	RunComponent(interpreters[nID - ID_FILE_INTERPRET1]);
 }
 
+void CGMEApp::OnRunComponentHelp(UINT nID) {
+	UINT id = (nID - ID_HELP_COMPONENTHELP_FIRST) / 2;
+	bool isEven = (nID - ID_HELP_COMPONENTHELP_FIRST) % 2 == 0;
+	CGMEEventLogger::LogGMEEvent(_T("CGMEApp::OnRunComponentHelp ")
+		+ (id < plugins.GetSize()
+			? plugins[id]
+			: (id < plugins.GetSize() + interpreters.GetSize()\
+				? interpreters[id - plugins.GetSize()]
+				: addons[id - plugins.GetSize() - interpreters.GetSize()]))
+		+ _T("\r\n"));
+
+	// Focus must be killed to flush ObjectInspector and Browser
+	::SetFocus(NULL);
+
+	CString onlineHelpLocation =
+		(id < plugins.GetSize()
+			? pluginOnlineHelp[id]
+			: (id < plugins.GetSize() + interpreters.GetSize()
+				? interpreterOnlineHelp[id - plugins.GetSize()]
+				: addonOnlineHelp[id - plugins.GetSize() - interpreters.GetSize()]));
+	CString offlineHelpLocation =
+		(id < plugins.GetSize()
+			? pluginOfflineHelp[id]
+			: (id < plugins.GetSize() + interpreters.GetSize()
+				? interpreterOfflineHelp[id - plugins.GetSize()]
+				: addonOfflineHelp[id - plugins.GetSize() - interpreters.GetSize()]));
+	ShellExecute(NULL, L"open"
+		, (isEven ? onlineHelpLocation : offlineHelpLocation)
+		, NULL, NULL, SW_SHOWNORMAL);
+	//RunComponent(interpreters[nID - ID_FILE_INTERPRET1]);
+}
+
+void CGMEApp::OnRunParadigmHelp(UINT nID)
+{
+	CComBSTR pname;
+	CComBSTR pver;
+	{
+		CComPtr<IMgaTerritory> t;
+		COMTHROW(mgaProject->CreateTerritory(NULL, &t, NULL));
+		COMTHROW(mgaProject->BeginTransaction(t, TRANSACTION_READ_ONLY));
+		COMTHROW(mgaProject->get_MetaName(&pname));
+		COMTHROW(mgaProject->get_MetaVersion(&pver));
+		COMTHROW(mgaProject->CommitTransaction());
+	}
+	std::wstringstream wss;

```

Comment:
This is weird. Wouldn't `static_cast<const wchar_t*>(pname)` work here?

Reply:
I just tried that. visual studio complained: "expression must have integral or unscoped enum type"
I found this works though:
CGMEEventLogger::LogGMEEvent((_T("CGMEApp::OnRunParadigmHelp ")
		+ std::wstring(static_cast<const wchar_t*>(pname))
		+ _T("\r\n")).c_str());

Reply:
I just commited&pushed the change